src/tests/string-case.ts

```diff
@@ -0,0 +1,73 @@
+import { doNotExecute, Expect, NotEqual, Equal } from "./utils";
+
+doNotExecute(async () => {
+  const string: "A" | "B" = "A";
+
+  const lowerCaseString = string.toLowerCase();
+
+  type tests = [Expect<NotEqual<typeof string, typeof lowerCaseString>>];
+});
+
+doNotExecute(async () => {
+  const string: "A" | "B" = "A";
```

Comment:
despite you put a type for a string variable, TS narrows it to "A". update it with `as` to get expected type
```suggestion
  const string = "A" as "A" | "B";
```